subprojects/build-init/src/testFixtures/groovy/org/gradle/buildinit/plugins/fixtures/ScriptDslFixture.groovy

```diff
@@ -99,6 +105,68 @@ class ScriptDslFixture {
         }
     }
 
+    private String blockTitleForSuite(suiteName) {
+        switch (scriptDsl) {
+            case KOTLIN:
+                String delegateName = (suiteName == JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME) ? "getting" : "registering"
+                return "val $suiteName by $delegateName(${JvmTestSuite.class.simpleName}::class)"
+            case GROOVY:
+                return suiteName
+            default:
+                throw new IllegalStateException("Unknown scriptDsl type: " + scriptDsl);
+        }
+    }
+
+    @Nullable
+    private String extractTopLevelBlockContents(String blockTitle, String text) {
+        def allLines = text.readLines()
+        def linesContainingBlock = allLines.findIndexValues { it.trim().startsWith(blockTitle) }
+        if (linesContainingBlock.size() != 1) {
+            return null;
+        }
+
+        int startIdx = linesContainingBlock[0] + 1;
+        int endIdx = allLines.size()
+
+        int openBraces = 1
+        for (int idx = startIdx; idx < allLines.size(); idx++) {
+            if (allLines[idx].contains("{")) {
+                openBraces++
+            }
+            if (allLines[idx].contains("}")) {
+                openBraces--
+            }
+            if (openBraces == 0) {
+                endIdx = idx - 1
+                break
+            }
+        }
+
+        if (endIdx == allLines.size()) {
+            return null
+        } else {
+            return allLines[startIdx..endIdx].join('\n').trim()
+        }
+    }
+
```

Comment:
I know it's just a test fixture and not production code, but do we really want to be parsing scripts? I see maintenance headaches. Let's talk options.

Reply:
Tabling for now